File: services/account/src/main/service/random-manager.js
import { RandomService } from './random-service.js';

class RandomManager extends RandomService {
  generateRandomString(pool, length) {
    let output = '';
    for (let i = 0; i < length; i++) {
      output += pool.charAt(Math.floor(Math.random() * pool.length));
    }
    return output;
  }
}

export {
  RandomManager
};

File: services/profile/src/main/java/base/profile/service/RandomManager.java
package base.profile.service;

public class RandomManager implements RandomService {
    @Override
    public String generateRandomString(String pool, int length) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < length; i++) {
            result.append(pool.charAt((int) (Math.random() * pool.length())));
        }
        return result.toString();
    }

}
